src/index.ts

```diff
@@ -45,7 +45,7 @@ export async function execute(filenames: string[], options: Partial<ExecutionOpt
         try {
             component.scripts = component.scripts.map((script: ComponentScript) => {
                 script.uri = path.join(
-                    options.root ? options.root : __dirname,
+                    options.root ? options.root : executionOptions.root,
```

Comment:
Based on this string:

    const executionOptions = Object.assign(defaultExecutionOptions, options);

is there a time that _root_ has something that isn't in _executionOptions_?  Should this instead be just _executionOptions.root_?

Reply:
Looks like `Object.assign` returns an object of type `ExecutionOptions & Partial<ExecutionOptions>` (basically `typeof A & typeof B`).  Switching that to `const executionOptions = { ...defaultExecutionOptions, options };` on line 38 should take care of that possibly-undefined `root` and allow this to be simplified to `executionoptions.root`!

Reply:
Thanks for answering Sean, forgot to check this yesterday.